### PR TITLE
Use bazelisk to download bazels for tests, instead of trying to download

### LINKED ISSUE #67: Support using <FORK>/<VERSION> in `.bazelversion` file.
It is possible to specify that an integration test uses the Bazel version specified in a `.bazelversion` file. It would be useful to support [Bazelisk's `<FORK>/<VERSION>`](https://github.com/bazelbuild/bazelisk#where-does-bazelisk-get-bazel-from).

### PR DESCRIPTION
**Pull request by katre**
directly.

Fixes #67.

### CODE REVIEW

**Comment by katre** [comment]
looks like I needed to test with bzlmod

**Comment by katre** [comment]
Ah, because I didn't update the bzlmod bazel_binaries extension.

File: bazel_integration_test/py/test_base.py [hunk 1]
```diff
@@ -252,8 +252,10 @@ def _EnvMap(self, env_remove = None, env_add = None):
       java_home = TestBase.GetEnv('JAVA_HOME', '')
       if java_home:
         env['JAVA_HOME'] = java_home
+      # TODO: figure out Windows runfiles_dir
```
Comment (cgrindel) [inline bazel_integration_test/py/test_base.py:255]: Let's create an issue and replace `TODO` with the `GHXXX` where `XXX` is the issue number.

File: bazel_integration_test/bzlmod/bazel_binaries.bzl [hunk 2]
```diff
@@ -118,14 +123,26 @@ version, version_file.\
         _bazel_binary_repo_rule(
             name = vi.repo_name,
             version_file = download.version_file,
+            bazelisk = "@%s//:bazelisk" % bazelisk_repo_name,
         )
     return vi
 
+# TODO: Make this configurable.
+_BAZELISK_VERSION = "1.18.0"
```
Comment (cgrindel) [inline bazel_integration_test/bzlmod/bazel_binaries.bzl:131]: I noticed that this version value is also hard-coded in the `bazel_binaries` macro. Do we need this value in both places?

**Review by cgrindel — COMMENTED** [review]
Overall, looks good. I left a couple of small comments in the code. We should also add an example that uses this new functionality and include it in the integration test suite.

This is more of a thought and not an action item. Have you thought about whether we could just use the `bazelisk` installed on the client's system, if it is detected? Typically, I am all about downloading the tools that we need. However, `bazelisk` is special in that it is the bootstrap for launching Bazel. 🤔

File: bazel_integration_test/py/test_base.py (hunk 1 above)
Comment (katre) [reply bazel_integration_test/py/test_base.py:255]: Broke this out to a new issue (#182) and a new PR (#183).

File: bazel_integration_test/bzlmod/bazel_binaries.bzl (hunk 2 above)
Comment (katre) [reply bazel_integration_test/bzlmod/bazel_binaries.bzl:131]: In the non-bzlmod version, this is an parameter to the `bazel_binaries` macro which can be overriden but defaults to 1.18.0. I couldn't figure out the syntax for bzlmod extensions (I think it needs a separate tag?), so it's just hardcoded here. Let me know if you'd prefer me to figure that out.

**Comment by katre** [comment]
As far as integration tests, all the tests and examples are now using this. Do we need further tests? I was trying to not add any behavior differences.

Using the local bazelisk sounds very similar to using a local bazel, maybe this is part of #100?

**Comment by cgrindel** [comment]
> As far as integration tests, all the tests and examples are now using this. Do we need further tests? I was trying to not add any behavior differences.

That is fair. Do we need any doc updates to let folks know that they can use the full range of Bazel version specifications that Bazelisk supports? In other words, how do you want to advertise this new capability?

> Using the local bazelisk sounds very similar to using a local bazel, maybe this is part of https://github.com/bazel-contrib/rules_bazel_integration_test/issues/100?

Maybe. We do not need to worry about using the local `bazelisk` for this PR.

File: bazel_integration_test/bzlmod/bazel_binaries.bzl (hunk 2 above)
Comment (cgrindel) [reply bazel_integration_test/bzlmod/bazel_binaries.bzl:131]: Could we create an issue and mark this with a `GHXXX` comment? I think that would be sufficient for now.

Comment (katre) [reply bazel_integration_test/bzlmod/bazel_binaries.bzl:131]: Done, filed #184.

**Comment by katre** [comment]
Rebased to main and added documentation (and a TODO issue for the bazelisk version).

**Review by cgrindel — APPROVED** [review]

**Comment by katre** [comment]
Any chance this can be added to a release?

**Pull request merged**